unity-client/Assets/Scripts/MainScripts/DCL/Helpers/MouseCatcher/MouseCatcher.cs

```diff
@@ -40,20 +44,41 @@ public void LockCursor()
             if (!renderingEnabled) return;
 
             Utils.LockCursor();
+#if !WEB_PLATFORM            
             OnMouseLock?.Invoke();
+#endif
         }
 
-        //Externally called by the browser
         public void UnlockCursor()
         {
             Utils.UnlockCursor();
+#if !WEB_PLATFORM            
             OnMouseUnlock?.Invoke();
```

Comment:
This means the events will not fire after a lock change in build? This isn't affecting any of our systems?

Reply:
> This means the events will not fire after a lock change in build? This isn't affecting any of our systems?

For web build the event was moved to the function called by kernel

        //Externally -ONLY- called by the browser
        public void UnlockCursorBrowser(int val)
        {
            bool lockPointer = val != 0;
            Utils.BrowserSetCursorState(lockPointer);
            if (lockPointer)
            {
                OnMouseLock?.Invoke();
            }
            else
            {
                OnMouseUnlock?.Invoke();
            }
        }

Reply:
We are triggering the mouse lock internally too, and `TaskbarHudController` depends on the event to trigger correctly on all cases. This may introduce bugs. (going to test it now to make sure)

Reply:
Ok nvm, I suppose that when we trigger the mouse lock internally, the browser event triggers and the roundtrip is completed